src/main/java/com/cleanroommc/groovyscript/compat/mods/astralsorcery/ChaliceInteraction.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    public void removeByOutput(ItemStack output) {
+        LinkedList<hellfirepvp.astralsorcery.common.base.LiquidInteraction> rts = (LinkedList<hellfirepvp.astralsorcery.common.base.LiquidInteraction>) LiquidInteractionAccessor.getRegisteredInteractions();
+
+        if (rts == null) return;
+        rts.forEach(rec -> {
+            if (((LiquidInteractionAccessor) rec).getFluidInteractionAction().getOutputForMatching().isItemEqual(output)) addBackup(rec);
+        });
+        rts.removeIf(rec -> ((LiquidInteractionAccessor) rec).getFluidInteractionAction().getOutputForMatching().isItemEqual(output));
+    }
+
+    public static RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    public static class RecipeBuilder extends AbstractRecipeBuilder<hellfirepvp.astralsorcery.common.base.LiquidInteraction> {
+
+        private final ArrayList<Float> chances = new ArrayList<>();
+        private ItemStack result;
+        private final ArrayList<FluidStack> components = new ArrayList<>();
+        private int probability = 1;
+
+        public RecipeBuilder result(ItemStack item) {
+            this.result = item;
+            return this;
+        }
+
+        public RecipeBuilder component(FluidStack fluid) {
+            this.components.add(fluid);
+            return this;
+        }
+
+        public RecipeBuilder component(FluidStack fluid1, FluidStack fluid2) {
+            this.components.add(fluid1);
+            this.components.add(fluid2);
+            return this;
+        }
+
+        public RecipeBuilder weight(int weight) {
+            this.probability = weight;
+            return this;
+        }
+
+        public RecipeBuilder chance(float consumptionChance) {
+            this.chances.add(consumptionChance);
+            return this;
+        }
+
+        public RecipeBuilder chance(float consumptionChance1, float consumptionChance2) {
+            this.chances.add(consumptionChance1);
+            this.chances.add(consumptionChance2);
+            return this;
+        }
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding Astral Sorcery Liquid Interaction";
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            msg.add(this.probability > 0, () -> "Weight must be a positive integer. Instead found: " + this.probability);
+            msg.add(this.chances.size() != 2, () -> "Exactly two consumption chances must be provided. Number provided: " + this.chances.size());
+            msg.add(this.chances.get(0) < 0 || this.chances.get(1) < 0, () -> "Consumption chance cannot be negative");
+            msg.add(this.components.size() != 2, () -> "Exactly two fluids must be provided. Number provided: " + this.components.size());
+            msg.add(this.components.get(0) == null || this.components.get(1) == null, () -> "Neither fluid can be null");
```

Comment:
I don't like how this is done. It creates somewhat ugly result script.
I suggest to have a `input(FluidStack fluid, float chance)` and a `input(FluidStack fluid)`, which calls `input(fluid, 1.0f)`.
This ensures that input list and chances list both have the same size.
Also for micro optimization you can change `ArrayList<Float>` to `FloatArrayList` from fastutils.